### PR TITLE
Update cocoon backend doc

### PR DESCRIPTION
**Pull request by keyonghan**
This is to address: https://github.com/flutter/flutter/issues/68337

### CODE REVIEW

File: app_dart/README.md [hunk 1]
```diff
@@ -4,19 +4,24 @@ This folder contains a Dart based backend for Cocoon.
...
+gcloud auth login
 gcloud init
 ```
+* Install Flutter
+```sh
+export PATH="$PATH":"path/to/flutter/bin/"
+flutter upgrade
+flutter pub get
+export PATH="$PATH":"path/to/flutter/bin/cache/dart-sdk/bin/"
+flutter config --enable-web
```
Comment (CaseyHillers) [inline app_dart/README.md:23]: This should no longer be needed as it's moved into stable

File: app_dart/README.md [hunk 2]
```diff
@@ -4,19 +4,24 @@ This folder contains a Dart based backend for Cocoon.
...
-Upon downloading and installing the Google Cloud SDK, you should initialize it
+* Install the Google Cloud Developer Tools Command Line Interface
+([`gcloud`](https://cloud.google.com/sdk/docs/quickstarts)). Then initialize it
 and authenticate yourself by running:
 
 ```sh
+gcloud auth login
 gcloud init
 ```
+* Install Flutter
```
Comment (CaseyHillers) [inline app_dart/README.md:17]: Should we just redirect to https://flutter.dev/docs/get-started/install ? We skipped this at first when web was experimental and required the config flag below.

File: app_dart/README.md [hunk 3]
```diff
@@ -80,13 +85,30 @@ To update the cron tasks in the App Engine project, run:
...
 
-### ssh into instance
+### Local development
+
+#### Using physical machine
+
+* Starting server
+
+```sh
+export COCOON_USE_IN_MEMORY_CACHE=true
```
Comment (CaseyHillers) [inline app_dart/README.md:95]: This should be a part of setting up the environment.

File: app_dart/README.md [hunk 4]
```diff
@@ -80,13 +85,30 @@ To update the cron tasks in the App Engine project, run:
...
+dart bin/server.dart
+```
+
+If you see Serving requests at 0.0.0.0:8080 the dev server is working.
+You should also set `COCOON_USE_IN_MEMORY_CACHE=true` as you typically
```
Comment (CaseyHillers) [inline app_dart/README.md:100]: > typically

Do we ever have access to redis?

File: app_dart/README.md [hunk 5]
```diff
@@ -80,13 +85,30 @@ To update the cron tasks in the App Engine project, run:
...
+don't have access to the remote redis instance during local development.
+
+#### Using Docker
+
+* ssh into instance
```
Comment (CaseyHillers) [inline app_dart/README.md:105]: Should this be "Starting docker instance"?

File: app_dart/README.md [hunk 6]
```diff
@@ -80,13 +85,30 @@ To update the cron tasks in the App Engine project, run:
...
 
 ```sh
 $ docker exec -it <container name> /bin/bash
 ```
 
-### Running a local development instance
+* Running a local development instance
```
Comment (CaseyHillers) [inline app_dart/README.md:111]: Should this be before the "ssh into instance" section?

File: app_dart/README.md [hunk 7]
```diff
@@ -111,6 +133,22 @@ $ docker images|grep local|tr -s ' '|cut -d' ' -f3|xargs docker rmi -f
 
 ### Deploying a release to App Engine
 
+#### Auto-deploy
+Cocoon auto deployment has been set up via
+[Google Cloud Build](https://pantheon.corp.google.com/cloud-build/triggers?project=flutter-dashboard)
+daily on Workdays. Please view [details](https://g3doc.corp.google.com/company/teams/flutter/cocoon/cloud_deploy.md?cl=head)
```
Comment (CaseyHillers) [inline app_dart/README.md:139]: edge case: This is probably fine to help with visibility, but I think having something like a go link would be preferable

File: app_dart/README.md [hunk 8]
```diff
@@ -111,6 +133,22 @@ $ docker images|grep local|tr -s ' '|cut -d' ' -f3|xargs docker rmi -f
 
 ### Deploying a release to App Engine
 
+#### Auto-deploy
+Cocoon auto deployment has been set up via
+[Google Cloud Build](https://pantheon.corp.google.com/cloud-build/triggers?project=flutter-dashboard)
```
Comment (CaseyHillers) [inline app_dart/README.md:138]: nit: avoid corp links

File: app_dart/README.md [hunk 9]
```diff
@@ -111,6 +133,22 @@ $ docker images|grep local|tr -s ' '|cut -d' ' -f3|xargs docker rmi -f
...
+daily on Workdays. Please view [details](https://g3doc.corp.google.com/company/teams/flutter/cocoon/cloud_deploy.md?cl=head)
+about cloud deploy setup.
+
+#### Manual-deploy
+
+* Using the cloud build
+
+This is easy to deploy if you simply want a new version based on
+the latest master commit. Open
+[Cloud Build dahsboard](https://pantheon.corp.google.com/cloud-build/triggers?project=flutter-dashboard)
```
Comment (CaseyHillers) [inline app_dart/README.md:148]: dahsboard => dashboard

File: app_dart/README.md [hunk 10]
```diff
@@ -111,6 +133,22 @@ $ docker images|grep local|tr -s ' '|cut -d' ' -f3|xargs docker rmi -f
 
 ### Deploying a release to App Engine
 
+#### Auto-deploy
+Cocoon auto deployment has been set up via
+[Google Cloud Build](https://pantheon.corp.google.com/cloud-build/triggers?project=flutter-dashboard)
+daily on Workdays. Please view [details](https://g3doc.corp.google.com/company/teams/flutter/cocoon/cloud_deploy.md?cl=head)
+about cloud deploy setup.
+
+#### Manual-deploy
```
Comment (CaseyHillers) [inline app_dart/README.md:142]: Should this link to the internal doc?

**Review by CaseyHillers — APPROVED** [review]
Thanks Keyong, LGTM when comments are addressed.

File: app_dart/README.md (hunk 10 above)
Comment (keyonghan) [reply app_dart/README.md:142]: Added.

File: app_dart/README.md (hunk 9 above)
Comment (keyonghan) [reply app_dart/README.md:148]: done.

File: app_dart/README.md (hunk 8 above)
Comment (keyonghan) [reply app_dart/README.md:138]: Good catch!

File: app_dart/README.md (hunk 7 above)
Comment (keyonghan) [reply app_dart/README.md:139]: Used go link instead.

File: app_dart/README.md (hunk 6 above)
Comment (keyonghan) [reply app_dart/README.md:111]: Switched.

File: app_dart/README.md (hunk 5 above)
Comment (keyonghan) [reply app_dart/README.md:105]: This is not starting the instance.

File: app_dart/README.md (hunk 4 above)
Comment (keyonghan) [reply app_dart/README.md:100]: Updated accordingly.

File: app_dart/README.md (hunk 3 above)
Comment (keyonghan) [reply app_dart/README.md:95]: Added a set up environment bullet.

File: app_dart/README.md (hunk 2 above)
Comment (keyonghan) [reply app_dart/README.md:17]: Added the link.
I received complains about missing `flutter upgrade` and `flutter pub get` when people refer to the internal doc to develop locally. So prefer to keep them here.

File: app_dart/README.md (hunk 1 above)
Comment (keyonghan) [reply app_dart/README.md:23]: Remoded.

**Pull request merged**